<!DOCTYPE html>
<html >

<head>

  <meta charset="UTF-8">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <title>Chapter 5 Collect | R for Environmental Health Research</title>
  <meta name="description" content="Online booklet for workshop on R for environmental health research, geared to those who already use R for their research but want to learn more tools.">
  <meta name="generator" content="bookdown  and GitBook 2.6.7">

  <meta property="og:title" content="Chapter 5 Collect | R for Environmental Health Research" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="Online booklet for workshop on R for environmental health research, geared to those who already use R for their research but want to learn more tools." />
  <meta name="github-repo" content="rstudio/bookdown-demo" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 5 Collect | R for Environmental Health Research" />
  
  <meta name="twitter:description" content="Online booklet for workshop on R for environmental health research, geared to those who already use R for their research but want to learn more tools." />
  

<meta name="author" content="Brooke Anderson">


<meta name="date" content="2019-04-09">

  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black">
  
  
<link rel="prev" href="package.html">
<link rel="next" href="process.html">
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />









<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<link rel="stylesheet" href="style.css" type="text/css" />
</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li><a href="./">Data Visualization in R</a></li>

<li class="divider"></li>
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Prerequisites</a><ul>
<li class="chapter" data-level="1.1" data-path="index.html"><a href="index.html#set-up"><i class="fa fa-check"></i><b>1.1</b> Set-up</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="organize.html"><a href="organize.html"><i class="fa fa-check"></i><b>2</b> Organize</a><ul>
<li class="chapter" data-level="2.1" data-path="organize.html"><a href="organize.html#r-projects"><i class="fa fa-check"></i><b>2.1</b> R Projects</a></li>
<li class="chapter" data-level="2.2" data-path="organize.html"><a href="organize.html#directory-organization"><i class="fa fa-check"></i><b>2.2</b> Directory organization</a><ul>
<li class="chapter" data-level="2.2.1" data-path="organize.html"><a href="organize.html#keeping-things-tidy"><i class="fa fa-check"></i><b>2.2.1</b> Keeping things tidy</a></li>
<li class="chapter" data-level="2.2.2" data-path="organize.html"><a href="organize.html#avoiding-repetition"><i class="fa fa-check"></i><b>2.2.2</b> Avoiding repetition</a></li>
</ul></li>
<li class="chapter" data-level="2.3" data-path="organize.html"><a href="organize.html#learn-more"><i class="fa fa-check"></i><b>2.3</b> Learn more</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="track.html"><a href="track.html"><i class="fa fa-check"></i><b>3</b> Track</a><ul>
<li class="chapter" data-level="3.1" data-path="track.html"><a href="track.html#git"><i class="fa fa-check"></i><b>3.1</b> git</a><ul>
<li class="chapter" data-level="3.1.1" data-path="track.html"><a href="track.html#configuring-git"><i class="fa fa-check"></i><b>3.1.1</b> Configuring git</a></li>
<li class="chapter" data-level="3.1.2" data-path="track.html"><a href="track.html#tracking-an-r-project"><i class="fa fa-check"></i><b>3.1.2</b> Tracking an R project</a></li>
</ul></li>
<li class="chapter" data-level="3.2" data-path="track.html"><a href="track.html#github"><i class="fa fa-check"></i><b>3.2</b> GitHub</a></li>
<li class="chapter" data-level="3.3" data-path="track.html"><a href="track.html#terminal"><i class="fa fa-check"></i><b>3.3</b> Terminal</a></li>
<li class="chapter" data-level="3.4" data-path="track.html"><a href="track.html#try-it-out"><i class="fa fa-check"></i><b>3.4</b> Try it out</a><ul>
<li class="chapter" data-level="3.4.1" data-path="track.html"><a href="track.html#track-a-project-with-git"><i class="fa fa-check"></i><b>3.4.1</b> Track a project with git</a></li>
<li class="chapter" data-level="3.4.2" data-path="track.html"><a href="track.html#link-the-project-with-a-github-repository"><i class="fa fa-check"></i><b>3.4.2</b> Link the project with a GitHub repository</a></li>
</ul></li>
<li class="chapter" data-level="3.5" data-path="track.html"><a href="track.html#learn-more-1"><i class="fa fa-check"></i><b>3.5</b> Learn more</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="package.html"><a href="package.html"><i class="fa fa-check"></i><b>4</b> Package</a><ul>
<li class="chapter" data-level="4.1" data-path="package.html"><a href="package.html#components-of-r-packages"><i class="fa fa-check"></i><b>4.1</b> Components of R packages</a><ul>
<li class="chapter" data-level="4.1.1" data-path="package.html"><a href="package.html#r-code"><i class="fa fa-check"></i><b>4.1.1</b> R code</a></li>
<li class="chapter" data-level="4.1.2" data-path="package.html"><a href="package.html#description-file"><i class="fa fa-check"></i><b>4.1.2</b> DESCRIPTION file</a></li>
<li class="chapter" data-level="4.1.3" data-path="package.html"><a href="package.html#other-package-components"><i class="fa fa-check"></i><b>4.1.3</b> Other package components</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="package.html"><a href="package.html#try-it-out-1"><i class="fa fa-check"></i><b>4.2</b> Try it out</a></li>
<li class="chapter" data-level="4.3" data-path="package.html"><a href="package.html#sharing"><i class="fa fa-check"></i><b>4.3</b> Sharing</a><ul>
<li class="chapter" data-level="4.3.1" data-path="package.html"><a href="package.html#when-to-publish-your-packages"><i class="fa fa-check"></i><b>4.3.1</b> When to publish your packages</a></li>
<li class="chapter" data-level="4.3.2" data-path="package.html"><a href="package.html#where-to-publish-packages"><i class="fa fa-check"></i><b>4.3.2</b> Where to publish packages</a></li>
<li class="chapter" data-level="4.3.3" data-path="package.html"><a href="package.html#extra-steps-for-publishing"><i class="fa fa-check"></i><b>4.3.3</b> Extra steps for publishing</a></li>
</ul></li>
<li class="chapter" data-level="4.4" data-path="package.html"><a href="package.html#learn-more-2"><i class="fa fa-check"></i><b>4.4</b> Learn more</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="collect.html"><a href="collect.html"><i class="fa fa-check"></i><b>5</b> Collect</a><ul>
<li class="chapter" data-level="5.1" data-path="collect.html"><a href="collect.html#open-data"><i class="fa fa-check"></i><b>5.1</b> Open data</a><ul>
<li class="chapter" data-level="5.1.1" data-path="collect.html"><a href="collect.html#admin-databases"><i class="fa fa-check"></i><b>5.1.1</b> [Admin databases?]</a></li>
<li class="chapter" data-level="5.1.2" data-path="collect.html"><a href="collect.html#research-repositories"><i class="fa fa-check"></i><b>5.1.2</b> [Research repositories?]</a></li>
</ul></li>
<li class="chapter" data-level="5.2" data-path="collect.html"><a href="collect.html#data-packages"><i class="fa fa-check"></i><b>5.2</b> Data packages</a></li>
<li class="chapter" data-level="5.3" data-path="collect.html"><a href="collect.html#web-services"><i class="fa fa-check"></i><b>5.3</b> Web services [?]</a></li>
<li class="chapter" data-level="5.4" data-path="collect.html"><a href="collect.html#ropensci"><i class="fa fa-check"></i><b>5.4</b> ROpenSci</a></li>
<li class="chapter" data-level="5.5" data-path="collect.html"><a href="collect.html#learn-more-3"><i class="fa fa-check"></i><b>5.5</b> Learn more</a></li>
</ul></li>
<li class="chapter" data-level="6" data-path="process.html"><a href="process.html"><i class="fa fa-check"></i><b>6</b> Process</a><ul>
<li class="chapter" data-level="6.1" data-path="process.html"><a href="process.html#learn-more-4"><i class="fa fa-check"></i><b>6.1</b> Learn more</a></li>
</ul></li>
<li class="chapter" data-level="7" data-path="final-words.html"><a href="final-words.html"><i class="fa fa-check"></i><b>7</b> Final Words</a></li>
<li class="divider"></li>
<li><a href="https://github.com/rstudio/bookdown" target="blank">Published with bookdown</a></li>

</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">R for Environmental Health Research</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="collect" class="section level1">
<h1><span class="header-section-number">Chapter 5</span> Collect</h1>
<p><span class="newthought">For Environmental Health researchers, I think</span> one of the most exciting
developments in R recently is how it is changing how we can collect data, both for
exposures and outcomes. One direction for this development is how researchers can
collect and measure original data from experiments, including through new measurement
technologies (e.g., phone-based Apps) and through new or rapidly changing health-related
measurements (e.g., metabolomics, flow cytometry) and associated open-source
software.</p>
<p>R is also facilitating and leveraging rapid developments in how researchers can access and
query secondary data, including from [large admin databases?] and [data repositories
encouraged or required for some NIH-funded projects]. … This section will provide an
introduction to some of the ideas and techniques behind these developments for
collecting secondary or public-use data for environmental health research, as well as
give you somee directions on where to go to find R packages that facilitate collecting
open data from R.</p>
<div id="open-data" class="section level2">
<h2><span class="header-section-number">5.1</span> Open data</h2>
<p>A range of environmental datasets are available online, especially through national agencies.
For example, [NOAA] provides various weather datasets, while USGS has data on water quality,
[others]</p>
<p>You can visit webpages hosted by these agencies where you can download the datasets you need.
However, this process can become tedious if you need lots of datasets, as may be the case for
large studies incorporating many cities. Further, downloading the datasets “by hand” is hard
to make reproducible, unless you meticiously write down all the steps you took as you visited
the website. This means that your process will be harder for you to repeat in the future
or for others to replicate.</p>
<p>A growing collection of R packages are now available that allow you to download datasets
available online directly from R. This means that you can write an R script for your data
collection, making this step both better documented and more reproducible. …</p>
<p>Further, there are now a number of R packages that provide access to open datasets that the
package maintainer collected and processed and is now making available as an R package.
[More on R data packages.]</p>
<div id="admin-databases" class="section level3">
<h3><span class="header-section-number">5.1.1</span> [Admin databases?]</h3>
<p>Data collected by government groups like EPA, NOAA, USGS.</p>
</div>
<div id="research-repositories" class="section level3">
<h3><span class="header-section-number">5.1.2</span> [Research repositories?]</h3>
<p>Here, especially health-related data.
NIH-supported (e.g., Metabolomics Workbench).
NHANES?</p>
</div>
</div>
<div id="data-packages" class="section level2">
<h2><span class="header-section-number">5.2</span> Data packages</h2>
<p>…</p>
<p>This package is too big for CRAN to host, so we have it posted through our own package
repository.<a href="#fn16" class="footnote-ref" id="fnref16"><sup>16</sup></a> Because of this, you’ll need to take a few extra steps
to download and install the package:<a href="#fn17" class="footnote-ref" id="fnref17"><sup>17</sup></a></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(drat)
<span class="kw">addRepo</span>(<span class="st">&quot;geanders&quot;</span>)
<span class="kw">install.packages</span>(<span class="st">&quot;hurricaneexposuredata&quot;</span>)
<span class="kw">install.packages</span>(<span class="st">&quot;hurricaneexposure&quot;</span>)</code></pre>
<p>Once you’ve installed both the <code>hurricaneexposuredata</code> and <code>hurricaneexposure</code> packages, you
can load them as usual to access both the data and some functions to work with them:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">library</span>(hurricaneexposuredata)
<span class="kw">library</span>(hurricaneexposure)</code></pre>
<p>The <code>hurricaneexposure</code> package has a series of functions that let you explore different
exposures during storms. For example, to get the storms where either New York County, NY,
or Suffolk County, MA, (which includes Boston) were exposed to tropical storm-level winds
(17.5 m/s or higher), you can run:<a href="#fn18" class="footnote-ref" id="fnref18"><sup>18</sup></a></p>
<pre class="sourceCode r"><code class="sourceCode r">knitr<span class="op">::</span><span class="kw">include_graphics</span>(<span class="st">&quot;images/tab_completion_example.png&quot;</span>)</code></pre>
<p><img src="images/tab_completion_example.png" width="492" /></p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">county_wind</span>(<span class="dt">counties =</span> <span class="kw">c</span>(<span class="st">&quot;36061&quot;</span>, <span class="st">&quot;25025&quot;</span>),
            <span class="dt">start_year =</span> <span class="dv">1988</span>, <span class="dt">end_year =</span> <span class="dv">2015</span>, 
            <span class="dt">wind_limit =</span> <span class="fl">17.5</span>)</code></pre>
<pre><code>##       storm_id  fips vmax_sust vmax_gust sust_dur gust_dur
## 1     Bob-1991 25025  26.46639  39.43492      210      570
## 2     Bob-1991 36061  18.19559  27.11142        0      480
## 3  Bertha-1996 25025  29.64453  44.17035      240      525
## 4  Bertha-1996 36061  28.95496  43.14289      180      540
## 5   Floyd-1999 25025  24.46946  36.45949      345      750
## 6   Floyd-1999 36061  20.50178  30.54765       60      315
## 7   Hanna-2008 25025  18.36505  27.36392        0      150
## 8   Hanna-2008 36061  19.25390  28.68832        0      195
## 9   Irene-2011 36061  25.68553  38.27144      165      510
## 10  Sandy-2012 36061  21.99213  32.76827      225      795
##    closest_time_utc storm_dist       local_time closest_date
## 1  1991-08-19 20:00  27.042565 1991-08-19 16:00   1991-08-19
## 2  1991-08-19 15:00 161.571830 1991-08-19 11:00   1991-08-19
## 3  1996-07-14 01:15  38.177990 1996-07-13 21:15   1996-07-13
## 4  1996-07-13 19:45  16.966013 1996-07-13 15:45   1996-07-13
## 5  1999-09-17 07:45  51.254726 1999-09-17 03:45   1999-09-17
## 6  1999-09-17 00:15  45.408483 1999-09-16 20:15   1999-09-16
## 7  2008-09-07 07:15   6.202866 2008-09-07 03:15   2008-09-07
## 8  2008-09-07 01:45  29.916672 2008-09-06 21:45   2008-09-06
## 9  2011-08-28 13:15   5.796733 2011-08-28 09:15   2011-08-28
## 10 2012-10-30 00:30 158.040788 2012-10-29 20:30   2012-10-29</code></pre>
<p>If you look up events based on flood events, you can instead run the
<code>county_events</code> function:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">county_events</span>(<span class="dt">counties =</span> <span class="kw">c</span>(<span class="st">&quot;36061&quot;</span>, <span class="st">&quot;25025&quot;</span>),
             <span class="dt">start_year =</span> <span class="dv">1988</span>, <span class="dt">end_year =</span> <span class="dv">2015</span>, 
             <span class="dt">event_type =</span> <span class="st">&quot;flood&quot;</span>)</code></pre>
<pre><code>##     fips     storm_id closest_time_utc storm_dist       local_time
## 1  25025  Dennis-1999 1999-09-08 08:00 390.047523 1999-09-08 04:00
## 2  36061   Floyd-1999 1999-09-17 00:15  45.408483 1999-09-16 20:15
## 3  36061 Allison-2001 2001-06-17 14:15 158.909890 2001-06-17 10:15
## 4  36061 Frances-2004 2004-09-09 13:30 379.343696 2004-09-09 09:30
## 5  36061    Ivan-2004 2004-09-18 17:45 311.346881 2004-09-18 13:45
## 6  36061  Jeanne-2004 2004-09-29 06:30 222.900157 2004-09-29 02:30
## 7  36061   Beryl-2006 2006-07-20 22:15 207.358443 2006-07-20 18:15
## 8  36061   Barry-2007 2007-06-04 15:45 148.251718 2007-06-04 11:45
## 9  36061   Irene-2011 2011-08-28 13:15   5.796733 2011-08-28 09:15
## 10 25025   Sandy-2012 2012-10-29 22:00 433.295980 2012-10-29 18:00
## 11 25025  Andrea-2013 2013-06-08 11:30  45.412565 2013-06-08 07:30
## 12 36061  Andrea-2013 2013-06-08 06:45  92.381282 2013-06-08 02:45
##    closest_date
## 1    1999-09-08
## 2    1999-09-16
## 3    2001-06-17
## 4    2004-09-09
## 5    2004-09-18
## 6    2004-09-29
## 7    2006-07-20
## 8    2007-06-04
## 9    2011-08-28
## 10   2012-10-29
## 11   2013-06-08
## 12   2013-06-08</code></pre>
<p>The <code>hurricaneexposure</code> package also has functions for mapping the exposure
data for specific storms. For example, to see the rainfall fro Hurricane Ivan in
2004, you can run:</p>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">map_counties</span>(<span class="dt">storm =</span> <span class="st">&quot;Ivan-2004&quot;</span>, <span class="dt">metric =</span> <span class="st">&quot;rainfall&quot;</span>)</code></pre>
<p><img src="columbia_env_health_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<p>From this map, you can see that the rain from this storm extended into New
England, even after the storm looped back around to the east and south. This is
why New York City had heavy rainfall (and flooding) from this event, but not
tropical storm-level winds.</p>
<p>For more on the <code>hurricaneexposure</code> package, see <a href="https://cran.r-project.org/web/packages/hurricaneexposure/vignettes/hurricaneexposure.html">its vignette</a>.</p>
</div>
<div id="web-services" class="section level2">
<h2><span class="header-section-number">5.3</span> Web services [?]</h2>
</div>
<div id="ropensci" class="section level2">
<h2><span class="header-section-number">5.4</span> ROpenSci</h2>
<p>One of the best places to explore R packages for accessing open data for science is
<a href="https://ropensci.org/"><strong>ROpenSci</strong></a>.<a href="#fn19" class="footnote-ref" id="fnref19"><sup>19</sup></a>
Many of its packages facilitate access to databases of open data relevant to scientific
research that have API access [?]. You can browse through its packages on
its <a href="https://ropensci.org/packages/"><strong>Packages</strong> page</a></p>
<p>Examples of some packages relevant to collecting data for environmental health research
include:</p>
<ul>
<li>bomrang: Australian Government Bureau of Meteorology (‘BOM’) Data Client</li>
<li>clifro: Easily Download and Visualise Climate Data from CliFlo</li>
<li>dbhydroR: ‘DBHYDRO’ Hydrologic and Water Quality Data</li>
<li>essurvey: Download Data from the European Social Survey on the Fly</li>
<li>FedData: Functions to Automate Downloading Geospatial Data Available from Several Federated Data Sources</li>
<li>camsRad: R Client for CAMS Radiation Service</li>
<li>ccafs: CCAFS GCM Data R Client</li>
<li>dbhydroR: R interface to the South Florida Water Management District’s DBHYDRO Database</li>
<li>biomartr: Genomic Data Retrieval</li>
<li>clifro: Easily Download and Visualise Climate Data from CliFlo</li>
<li>DataSpaceR: An R Interface to ‘the CAVD DataSpace’</li>
<li>essurvey: Download Data from the European Social Survey on the Fly</li>
<li>fingertipsR: Fingertips Data for Public Health</li>
<li>getCRUCLdata: Use and Explore ‘CRU’ ‘CL’ v. 2.0 Climatology Elements</li>
<li>getlandsat: Get Landsat 8 Data from Amazon Public Data Sets</li>
<li>googleLanguageR: Call Google’s ‘Natural Language’ API, ‘Cloud Translation’ API, ‘Cloud Speech’
API and ‘Cloud Text-to-Speech’ API</li>
<li>GSODR: Global Surface Summary of the Day (‘GSOD’) Weather Data Client</li>
<li>hydroscoper: Interface to the Greek National Data Bank for Hydrometeorological Information</li>
<li>MODIStsp: A Tool for Automating Download and Preprocessing of MODIS Land Products Data</li>
<li>nasapower: NASA POWER API Client</li>
<li>opencage: Interface to the OpenCage API</li>
<li>osmdata: Import ‘OpenStreetMap’ Data as Simple Features or Spatial Objects</li>
<li>weathercan: Download Weather Data from the Environment and Climate Change Canada Website</li>
<li>wateRinfo: Download Time Series Data from Waterinfo.be</li>
<li>USAboundariesData: Datasets for the ‘USAboundaries’ package</li>
<li>USAboundaries: Historical and Contemporary Boundaries of the United States of America</li>
<li>tidyhydat: Extract and Tidy Canadian ‘Hydrometric’ Data</li>
<li>stats19: Work with Open Road Traffic Casualty Data from Great Britain</li>
<li>smapr: Acquisition and Processing of NASA Soil Moisture Active-Passive (SMAP) Data</li>
<li>rWBclimate: A package for accessing World Bank climate data</li>
<li>rusda: Interface to USDA Databases</li>
<li>rsnps: Get ‘SNP’ (‘Single-Nucleotide’ ‘Polymorphism’) Data on the Web</li>
<li>rrricanesdata: Data for Atlantic and east Pacific tropical cyclones since 1998</li>
<li>rrricanes: Web scraper for Atlantic and east Pacific hurricanes and tropical storms</li>
<li>ropenaq: Accesses Air Quality Data from the Open Data Platform OpenAQ</li>
<li>rnoaa: ‘NOAA’ Weather Data from R</li>
<li>rnaturalearth: World Map Data from Natural Earth</li>
<li>riem: Accesses Weather Data from the Iowa Environment Mesonet</li>
<li>rgpdd: R Interface to the Global Population Dynamics Database</li>
<li>rdhs: API Client and Dataset Management for the Demographic and Health Survey (DHS) Data</li>
<li>rdefra: Interact with the UK AIR Pollution Database from DEFRA</li>
<li>prism: Access Data from the Oregon State Prism Climate Project</li>
</ul>
<p>Cleaning / exploring data:</p>
<ul>
<li>cleanEHR: The Critical Care Clinical Data Processing Tools</li>
<li>colocr: Conduct Co-localization Analysis of Fluorescence Microscopy Images</li>
<li>cRegulome: Obtain and Visualize Regulome-Gene Expression Correlations in Cancer</li>
<li>EndoMineR: Functions to mine endoscopic and associated pathology datasets</li>
<li>geoaxe: Split ‘Geospatial’ Objects into Pieces</li>
<li>hddtools: Hydrological Data Discovery Tools</li>
<li>isdparser: Parse ‘NOAA’ Integrated Surface Data Files</li>
<li>visdat: Preliminary Visualisation of Data</li>
<li>skimr: A frictionless, pipeable approach to dealing with summary statistics</li>
</ul>
</div>
<div id="learn-more-3" class="section level2">
<h2><span class="header-section-number">5.5</span> Learn more</h2>
<ul>
<li>ROpenSci</li>
<li>The R Journal</li>
<li>CRAN task views?</li>
<li>ROpenSci articles?</li>
<li>JOSS?</li>
</ul>

</div>
</div>
<div class="footnotes">
<hr />
<ol start="16">
<li id="fn16"><p>For more on how and why we did this, see
<a href="https://journal.r-project.org/archive/2017/RJ-2017-026/">an article</a> we wrote about the
process for <strong>The R Journal</strong>.<a href="collect.html#fnref16" class="footnote-back">↩</a></p></li>
<li id="fn17"><p>If you don’t have the <code>drat</code> package, install it in
the usual way with <code>install.packages("drat")</code>.<a href="collect.html#fnref17" class="footnote-back">↩</a></p></li>
<li id="fn18"><p>For any of these functions, you can find out what parameters
to include, and in what format, but opening the helpfile for the function. For example, once you’ve
loaded the <code>hurricaneexposure</code> package, you can open the helpfile for the <code>county_wind</code> function
by calling <code>?county_wind</code>. Also helpful for navigating packages: take advantage of the <code>package::function</code>
notation and RStudio’s tab completion to look up the names of functions in a package. For example,
type <code>hurricaneexposure::</code>. A pop-up window should show up with all the functions in the package (press
Tab if the pop-up doesn’t automatically open).<a href="collect.html#fnref18" class="footnote-back">↩</a></p></li>
<li id="fn19"><p><a href="https://ropensci.org/"><strong>ROpenSci</strong></a>. …<a href="collect.html#fnref19" class="footnote-back">↩</a></p></li>
</ol>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="package.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="process.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"google": false,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "google", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": "https://github.com/rstudio/bookdown-demo/edit/master/04-collect.Rmd",
"text": "Edit"
},
"history": {
"link": null,
"text": null
},
"download": ["columbia_env_health.pdf"],
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
